File: vertx-elasticsearch/src/main/java/fr/myprysm/vertx/elasticsearch/action/search/ClearScrollResponse.java
package fr.myprysm.vertx.elasticsearch.action.search;

import io.vertx.codegen.annotations.DataObject;
import io.vertx.core.json.JsonObject;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.Accessors;
import org.elasticsearch.rest.RestStatus;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Accessors(chain = true)
@DataObject(generateConverter = true)
public class ClearScrollResponse {

    private boolean succeeded;
    private int numFreed;
    private RestStatus status;

    public ClearScrollResponse(ClearScrollResponse other) {
        succeeded = other.succeeded;
        numFreed = other.numFreed;
        status = other.status;
    }

    /**
     * Build a new <code>ClearScrollResponse</code> from a <code>JsonObject</code>.
     *
     * @param json the <code>JsonObject</code>
     */
    public ClearScrollResponse(JsonObject json) {
        ClearScrollResponseConverter.fromJson(json, this);
    }

    /**
     * Transforms the <code>ClearScrollResponse</code> into a <code>JsonObject</code>.
     *
     * @return the <code>JsonObject</code>
     */
    public JsonObject toJson() {
        JsonObject json = new JsonObject();
        ClearScrollResponseConverter.toJson(this, json);
        return json;
    }
}

File: vertx-elasticsearch/src/main/kotlin/fr/myprysm/vertx/elasticsearch/kotlin/action/search/ClearScrollResponse.kt
package fr.myprysm.vertx.elasticsearch.kotlin.action.search

import fr.myprysm.vertx.elasticsearch.action.search.ClearScrollResponse
import org.elasticsearch.rest.RestStatus

fun ClearScrollResponse(
  numFreed: Int? = null,
  status: RestStatus? = null,
  succeeded: Boolean? = null): ClearScrollResponse = fr.myprysm.vertx.elasticsearch.action.search.ClearScrollResponse().apply {

  if (numFreed != null) {
    this.setNumFreed(numFreed)
  }
  if (status != null) {
    this.setStatus(status)
  }
  if (succeeded != null) {
    this.setSucceeded(succeeded)
  }
}

